sources/zafira-services/src/main/java/com/qaprosoft/zafira/service/AuthService.java

```diff
@@ -0,0 +1,74 @@
+package com.qaprosoft.zafira.service;
+
+import com.qaprosoft.zafira.models.db.User;
+import com.qaprosoft.zafira.service.exception.AuthException;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import static com.qaprosoft.zafira.service.exception.AuthException.AuthErrorDetail.ADMIN_CREDENTIALS_INVALID;
+import static com.qaprosoft.zafira.service.exception.AuthException.AuthErrorDetail.INVALID_USER_CREDENTIALS;
+import static com.qaprosoft.zafira.service.exception.AuthException.AuthErrorDetail.USER_BELONGS_TO_OTHER_TENANT;
+import static com.qaprosoft.zafira.service.exception.AuthException.AuthErrorDetail.USER_INACTIVE;
+
+@Service
+public class AuthService {
+
+    private static final String ERR_MSG_INVALID_CREDENTIALS = "Invalid credentials for user %s";
+    private static final String ERR_MSG_USER_IS_INACTIVE = "User id %d is inactive";
+    private static final String ERR_MSG_USER_IS_FROM_OTHER_TENANT = "User with id %d is from tenant %s";
+    private static final String ERR_MSG_USERNAME_OR_PASSWORD_IS_INVALID = "Username or password for user %s is invalid";
+
+    private final AuthenticationManager authenticationInternalManager;
+    private final AuthenticationManager authenticationLdapManager;
+    private final UserService userService;
+    private final String adminUsername;
+
+    public AuthService(AuthenticationManager authenticationInternalManager,
+                       AuthenticationManager authenticationLdapManager,
+                       UserService userService,
+                       @Value("${zafira.admin.username}") String adminUsername) {
+        this.authenticationInternalManager = authenticationInternalManager;
+        this.authenticationLdapManager = authenticationLdapManager;
+        this.userService = userService;
+        this.adminUsername = adminUsername;
+    }
+
+    public Authentication getAuthentication(String username, String password, User user) {
+        Authentication authentication;
+        try {
+            final AuthenticationManager authenticationManager = user == null
```

Comment:
Let's use if-else instead of ternary operator here for the sake of readability

Reply:
Looks like you've already fixed it